backend/src/main/java/com/github/lenaruppert/backend/controller/MongoUserController.java

```diff
@@ -0,0 +1,102 @@
+import com.github.lenaruppert.backend.repository.MongoUserRepository;
+import jakarta.servlet.http.HttpSession;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.security.Principal;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("api/user")
+@RequiredArgsConstructor
+public class MongoUserController {
+
+    private final MongoUserRepository mongoUserRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @PostMapping
+    public MongoUser create(@RequestBody MongoUserDTO user) {
+        if (user.username() == null || user.username().length() == 0) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Username is required");
+        }
+
+        if (user.password() == null || user.password().length() == 0) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Password is required");
+        }
+
+        if (mongoUserRepository.existsByUsername(user.username())) {
+            throw new ResponseStatusException(
+                    HttpStatus.CONFLICT,
+                    "User already exists"
+            );
+        }
+
+        MongoUser newUser = new MongoUser(
+                UUID.randomUUID().toString(),
+                user.username(),
+                passwordEncoder.encode(user.password()),
+                "BASIC"
+        );
+
+        MongoUser out = mongoUserRepository.save(newUser);
+
+        return new MongoUser(
+                out.id(),
+                out.username(),
+                null,
+                out.role()
+        );
+    }
+
+    @PostMapping("/login")
+    public MongoUser login(Principal principal) {
+        return getMe1(principal);
+    }
+
+    @PostMapping("/logout")
+    public void logout(HttpSession session) {
+        session.invalidate();
+        SecurityContextHolder.clearContext();
+    }
+
+    @GetMapping("/me")
+    public MongoUser getMe1(Principal principal) {
+        MongoUser me = mongoUserRepository
+                .findByUsername(principal.getName())
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.UNAUTHORIZED));
+
+        return new MongoUser(
+                me.id(),
+                me.username(),
+                null,
+                me.role()
+        );
+    }
+
+    @GetMapping("/me2")
+    public String getMe2() {
+        return SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getName();
+    }
+
+    @GetMapping("/admin")
+    public String getAdminStatus() {
+        return "Admin OK";
+    }
+
+    @GetMapping
+    public String getStatus() {
+        return "OK";
+    }
```

Comment:
Are those actually used at this point? If not, I would advise to delete them to keep your code tidy :)